core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushdownLimitIntoWindow.java

```diff
@@ -56,6 +57,7 @@ public PushdownLimitIntoWindow(Metadata metadata)
         this.rankFunctionId = metadata.resolveFunction(QualifiedName.of("rank"), ImmutableList.of()).getFunctionId();
         this.pattern = limit()
                 .matching(limit -> !limit.isWithTies() && limit.getCount() != 0 && limit.getCount() <= Integer.MAX_VALUE)
+                .with(isOrderSensitive().equalTo(false))
```

Comment:
This should be allowed if the ordering scheme is compatible with the ordering scheme of the window function

Reply:
Making this change.
Should there be a PushdownTopNIntoWindow rule along similar lines ?